### PR TITLE
Allow older and newer versions of libgpiod to work, Allow Pullups

### LINKED ISSUE #425: python-libgpiod supports pull up/down resistors so this should be implemented
Recent versions of python-libgpiod support setting pull up/down resistors, this could be supported by changing the relevant section of `microcontroller/generic_linux/libgpiod_pin.py` to something like the following:

	if pull is not None:
	    if pull == self.PULL_UP:
	        flags += gpiod.LINE_REQ_FLAG_BIAS_PULL_UP
	    elif pull == self.PULL_DOWN:
	        flags += gpiod.LINE_REQ_FLAG_BIAS_PULL_DOWN
	    else:
	        raise RuntimeError("Invalid pull for pin: %s" % self.id)

### LINKED ISSUE #571: gpiod.Chip should not use uppercase
Somebody reported this on discord:
```
# uname -r
5.10.102-meson64
# cat /etc/debian_version
bullseye/sid
# python
Python 3.8.10 (default, Nov 26 2021, 20:14:08)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import board
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/board.py", line 158, in <module>
    from adafruit_blinka.board.hardkernel.odroidc2 import *
  File "/usr/local/lib/python3.8/dist-packages/adafruit_blinka/board/hardkernel/odroidc2.py", line 6, in <module>
    from adafruit_blinka.microcontroller.amlogic.s905 import pin
  File "/usr/local/lib/python3.8/dist-packages/adafruit_blinka/microcontroller/amlogic/s905/pin.py", line 7, in <module>
    GPIO122 = Pin((0, 0))
  File "/usr/local/lib/python3.8/dist-packages/adafruit_blinka/microcontroller/generic_linux/libgpiod_pin.py", line 34, in __init__
    self._chip = gpiod.Chip(str(pin_id[0]), gpiod.Chip.OPEN_BY_NUMBER)
AttributeError: module 'gpiod' has no attribute 'Chip'
```
Looking at the gpiod repo it does look like it should be `gpiod.chip` in lowercase ?
Or could there be a different version of the python gpiod module that conflicts ?

### PR DESCRIPTION
**Pull request by makermelissa**
Fixes #571. Fixes #425. Tested on Odroid C2 with older and newer libgpiod.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,9 +4,10 @@
 
 repos:
 -   repo: https://github.com/python/black
-    rev: 20.8b1
+    rev: 22.1.0
```
Comment (dhalbert) [inline .pre-commit-config.yaml:7]: `Even newer version available; we are using it in CPy libs:
``suggestion
    rev: 22.3.0
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
Just a version thing. Rest lgtm!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (makermelissa) [reply .pre-commit-config.yaml]: Updated

**Review by dhalbert — APPROVED** [review]
:+1:

**Pull request merged**